### PR TITLE
Optimize the installation page

### PR DESCRIPTION
**Pull request by kerwin612**

### CODE REVIEW

**Comment by lunny** [comment]
Could you post a screenshot?

**Comment by kerwin612** [comment]
image

**Review by techknowlogick — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I would say the UI looks more ugly than before, these "installing" texts are just redundancy, and the sizes (font/image) are uncoordinated.

Since there is already an animation, I do not see why adding more sentences makes sense.

**Comment by kerwin612** [comment]
> I would say the UI looks more ugly than before, these "installing" texts are just redundancy, and the sizes (font/image) are uncoordinated.
> 
> Since there is already an animation, I do not see why adding more sentences makes sense.

image
Title, animation, description; not conflicting...

**Comment by wxiaoguang** [comment]
> Title, animation, description; not conflicting...

Your sample screenshot is well-designed and good-looking. And it does provide more details about each step, not just keep saying "installing" again and again.

File: templates/post-install.tmpl [hunk 1]
```diff
@@ -6,14 +6,17 @@
...
 							<div>
-								<img src="{{AssetUrlPrefix}}/img/loading.png" alt="{{ctx.Locale.Tr "loading"}}">
+								<img src="{{AssetUrlPrefix}}/img/loading.png" alt="{{ctx.Locale.Tr "install.installing"}}">
 							</div>
 						</div>
 					</div>
 					<div class="ui stackable middle very relaxed page grid">
 						<div class="sixteen wide center aligned centered column">
-							<p><a id="goto-user-login" href="{{AppSubUrl}}/user/login">{{ctx.Locale.Tr "loading"}}</a></p>
+							<p><a id="goto-user-login" href="{{AppSubUrl}}/user/login">{{ctx.Locale.Tr "install.installing_desc"}}...</a></p>
```
Comment (delvh) [inline templates/post-install.tmpl]: Please add the `...` to the translated string directly

**Review by delvh — APPROVED** [review]
LGTM after the short fix.

**Review by wxiaoguang — CHANGES_REQUESTED** [review]
No, it doesn't look good to me

**Comment by techknowlogick** [comment]
I apologize if this is getting into yak-shaving territory.

I think having both "installing" and "please wait" is important, as even though there is the animation, for screen-readers they won't see that animation. This way, they'll know there is something happening that they need to wait for.

**Comment by wxiaoguang** [comment]
> I apologize if this is getting into yak-shaving territory.
> 
> I think having both "installing" and "please wait" is important, as even though there is the animation, for screen-readers they won't see that animation. This way, they'll know there is something happening that they need to wait for.

At least, the page should be well-designed and good-looking. But not adding more and more uncoordinated elements.

Have you ever seen other websites or apps use such layout ? Small font, large icon, small font, keep repeating "installing", and visually misaligned?

**Comment by kerwin612** [comment]
l1
l2

@wxiaoguang Is this OK?

**Comment by wxiaoguang** [comment]
> @wxiaoguang Is this OK?

Yes, much better than before, looks good to me.

**Comment by kerwin612** [comment]
> > @wxiaoguang Is this OK?
> 
> Yes, much better than before, looks good to me. (maybe it needs to fine tune the icon size, but it should be easy to do so)

ok, icon resized slightly, I have commited.

**Comment by wxiaoguang** [comment]
Thank you very much for improving the page.

I did some tests on my side, and I think some HTML element could be optimized too since we do not need them.

And we can add `aria-hidden="true"` to the image, then no need to set "alt" or expose it to screen reader.

image

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**